### PR TITLE
how to set ENV in doc

### PR DESCRIPTION
**Pull request by hyuraku**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?
- [ ] Have you successfully run `brew man` locally and committed any changes?

-----
add how to set environment variables anywhere to https://docs.brew.sh/Formula-Cookbook because of https://github.com/Homebrew/brew/issues/8648

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
I think it would be good to document this but I don't think this addition as-is provides much clarify. Could you try a dedicated section or similar?

File: docs/Formula-Cookbook.md [hunk 1]
```diff
@@ -64,6 +64,8 @@ class Foo < Formula
                           "--disable-dependency-tracking",
                           "--disable-silent-rules",
                           "--prefix=#{prefix}"
+    # set ENV
+    ENV["VERSION"] = version.to_s
```
Comment (hyuraku) [inline docs/Formula-Cookbook.md:68]: https://github.com/Homebrew/homebrew-core/blob/c52f9b1d041d901adc84a51afed860adbffb401d/Formula/gh.rb#L23

**Comment by MikeMcQuaid** [comment]
Sorry, this still isn't what I'm looking for; you've edited the example that `brew create` outputs. Consider adding some prose or a code example to https://github.com/Homebrew/brew/blob/441044932c4d475f3ca0a9b646396ee31e88dc0d/docs/Formula-Cookbook.md#using-environment-variables instead.

**Comment by hyuraku** [comment]
@MikeMcQuaid 
> Consider adding some prose or a code example to https://github.com/Homebrew/brew/blob/441044932c4d475f3ca0a9b646396ee31e88dc0d/docs/Formula-Cookbook.md#using-environment-variables instead.

Thank you for the advice, that’s very helpful.

File: docs/Formula-Cookbook.md [hunk 2]
```diff
@@ -757,6 +757,8 @@ Firstly, the overall environment in which Homebrew runs is filtered to avoid env
 
 The second level of filtering removes sensitive environment variables (such as credentials like keys, passwords or tokens) to avoid malicious subprocesses obtaining them (<https://github.com/Homebrew/brew/pull/2524>). This has the effect of preventing any such variables from reaching a formula's Ruby code as they are filtered before it is called. The specific implementation can be seen in the [`ENV.clear_sensitive_environment!` method](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/extend/ENV.rb).
 
+Third you can set environment variables for each formula's install method like  (<https://github.com/Homebrew/homebrew-core/blob/fd9ad29f8e3ca9476f838ebb13794ddb7dafba00/Formula/gh.rb#L22>) if you need.
```
Comment (MikeMcQuaid) [inline docs/Formula-Cookbook.md:760]:
```suggestion
You can set environment variables in a formula's `install` method using `ENV["VARIABLE_NAME"] = "VALUE"`. An example can be seen in [the `gh` formula](https://github.com/Homebrew/homebrew-core/blob/fd9ad29f8e3ca9476f838ebb13794ddb7dafba00/Formula/gh.rb#L22).
```

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks again @hyuraku!

**Pull request merged**